src/firefly/js/tables/ui/ActiveTableButton.jsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+
+import React, {Component,PropTypes} from 'react';
+import sCompare from 'react-addons-shallow-compare';
+
+import {flux} from '../../Firefly.js';
+import {getTblInfoById, getActiveTableId} from '../TableUtil.js';
+
+/**
+ * This download button does 2 things:
+ * 1. track a table for selected rows, then changes style to reflect whether or not it can be clicked
+ * 2. when clicked, display a popup dialog of the given download options panel
+ *    or a message prompting the user to select rows first
+ */
+export class ActiveTableButton extends Component {
+
```

Comment:
I had a little trouble figuring out how this all worked.  The key is the onChanges function that takes a table_id and a highlighted row and return a boolean and a title.  I think this should be documented.